optaplanner-core/src/main/java/org/optaplanner/core/config/localsearch/LocalSearchPhaseConfig.java

```diff
@@ -258,6 +261,9 @@ protected LocalSearchForager buildForager(HeuristicConfigPolicy configPolicy) {
                     // Fast stepping algorithm
                     foragerConfig_.setAcceptedCountLimit(1);
                     break;
+                case GREAT_DELUGE:
+                    // Fast stepping algorithm
+                    foragerConfig_.setAcceptedCountLimit(4);
```

Comment:
Question: Why acceptedCountLimit 4? I doubt if the canonical Great Deluge paper suggest a "tournament factor" of 4.
In the past, I've overfitted some examples's benchmarks to use an acceptedCountLimit of 4 for SA (cloud balancing for one). This legacy still lingers in many of the benchmark configs. There is no reliable basis for this, except that on the first example where I did it, nurse rostering, it worked better than 1 for the 10 medium datasets, but a neglectable margin. I 've not confirmed this decently with other use cases.

Let's go with acceptedCountLimit 1, just like SA in this code, unless there is proof on at least 3 examples for at least 5 datasets each that 4 works better. @triceo Wdyt?

Reply:
btw Lukas, we have all these examples, all these datasets, all this algorithms and a benchmarker to mine this gold mine of research information. Maybe we should spend some time in doing that - and meanwhile clean up the benchmark configs to avoid magic numbers like acceptedCountLimit=4, tabuSize=11, etc in those benchmark configs - unless they can be proven. Furthermore, it might affect our defaults too (such as tabuSize=7)

Reply:
I agree we should not be making decisions unless we have solid factual basis for them.
Boris and I actually tried to mine our own datasets, and we started comparing local searches in various configs against each other - unfortunately, we hit a wall with SA, the config of which is too problem-specific to allow any generalizations.

Let's discuss this elsewhere. This PR already has 90+ comments. :-)

Reply:
Yes, it's great work though. I look forward to having this in our arsenal of algorithms. Thanks Boris keeping up with all these hard reviews :)